sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateTable.scala

```diff
@@ -76,14 +82,31 @@ class StateTable(
   override def properties(): util.Map[String, String] = Map.empty[String, String].asJava
 
   private def isValidSchema(schema: StructType): Boolean = {
```

Comment:
My proposal could handle both non-CDF and CDF altogether and this still needs a divergence and also every column has its own if or else if. Have you tried with my proposal?

Reply:
Sorry misread the code. It is indeed more elegant.